src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -201,28 +204,30 @@ private void SetGlobalErrorIfTrue(bool arg)
 
                 if (moduleBeingBuiltOpt != null && entryPoint != null && compilation.Options.OutputKind.IsApplication())
                 {
-                    moduleBeingBuiltOpt.SetPEEntryPoint(entryPoint, diagnostics);
+                    moduleBeingBuiltOpt.SetPEEntryPoint(entryPoint, diagnostics.DiagnosticBag);
                 }
             }
         }
 
         // Returns the MethodSymbol for the assembly entrypoint.  If the user has a Task returning main,
         // this function returns the synthesized Main MethodSymbol.
-        private static MethodSymbol GetEntryPoint(CSharpCompilation compilation, PEModuleBuilder moduleBeingBuilt, bool hasDeclarationErrors, bool emitMethodBodies, DiagnosticBag diagnostics, CancellationToken cancellationToken)
+        private static MethodSymbol GetEntryPoint(CSharpCompilation compilation, PEModuleBuilder moduleBeingBuilt, bool hasDeclarationErrors, bool emitMethodBodies, BindingDiagnosticBag diagnostics, CancellationToken cancellationToken)
         {
+            Debug.Assert(diagnostics.DiagnosticBag != null);
```

Comment:
Consider using either `is object`, or `RoslynDebug.Assert`. This is a pure null check in an unannotated method, and will cause warnings when nullable is eventually enabled. A simple change now makes it easier later.

Reply:
> Consider using either is object, or RoslynDebug.Assert. This is a pure null check in an unannotated method, and will cause warnings when nullable is eventually enabled. A simple change now makes it easier later.

This file has lots of asserts that will have to be changed when nullable is enabled here, for now I keep the same stile as used throughout the file.

---
In reply to: [362136392](https://github.com/dotnet/roslyn/pull/40560#discussion_r362136392) [](ancestors = 362136392)